File: Java/SI1132.java
import com.pi4j.io.i2c.I2CBus;
import com.pi4j.io.i2c.I2CDevice;
import com.pi4j.io.i2c.I2CFactory;
import java.io.IOException;

public class SI1132
{
	public static void main(String args[]) throws Exception
	{
		// Create I2CBus
		I2CBus bus = I2CFactory.getInstance(I2CBus.BUS_1);
		// Get I2C device, SI1132 I2C address is 0x60(96)
		I2CDevice device = bus.getDevice(0x60);
		
		// Enable UVindex measurement coefficients
		device.write(0x13, (byte)0x29);
		device.write(0x14, (byte)0x89);
		device.write(0x15, (byte)0x02);
		device.write(0x16, (byte)0x00);
		
		// Select PARAM_WR register
		// Enable UV, Visible, IR
		device.write(0x17, (byte)0xF0);
		// Select COMMAND register
		// Select CHLIST register in RAM
		device.write(0x18, (byte)(0x01 | 0xA0));
		Thread.sleep(10);
		// Select PARAM_RD register
		// Read 1 byte of data
		byte status = (byte)device.read(0x2E);
		
		// Select INT Output Enable register
		// INT pin driven low
		device.write(0x03, (byte)0x01);
		
		// Select ALS Interrupt Enable register
		// Assert INT pin whenever VIS or UV measurements are ready
		device.write(0x04, (byte)0x01);
		
		// Select HW_KEY register
		// Default value
		device.write(0x07, (byte)0x17);
		
		// Select PARAM_WR register
		// Small IR photodiode
		device.write(0x17, (byte)0x00);
		// Select COMMAND register
		// Select ALS_IR_ADCMUX register in RAM
		device.write(0x18, (byte)(0x0E | 0xA0));
		Thread.sleep(10);
		// Select PARAM_RD register
		// Read 1 byte of data
		status = (byte)device.read(0x2E);
		
		// Select PARAM_WR register
		// Set ADC Clock divided / 1
		device.write(0x17, (byte)0x00);
		// Select COMMAND register
		// Select ALS_IR_ADC_GAIN register in RAM
		device.write(0x18, (byte)(0x1E | 0xA0));
		Thread.sleep(10);
		// Select PARAM_RD register
		// Read 1 byte of data
		status = (byte)device.read(0x2E);
		
		// Select PARAM_WR register
		// Set 511 ADC Clock
		device.write(0x17, (byte)0x70);
		// Select COMMAND register
		// Select ALS_IR_ADC_COUNTER register in RAM
		device.write(0x18, (byte)(0x1D | 0xA0));
		Thread.sleep(10);
		// Select PARAM_RD register
		// Read 1 byte of data
		status = (byte)device.read(0x2E);
		
		// Select PARAM_WR register
		// Set ADC Clock divided / 1
		device.write(0x17, (byte)0x00);
		// Select COMMAND register
		// Select ALS_VIS_ADC_GAIN register in RAM
		device.write(0x18, (byte)(0x11 | 0xA0));
		Thread.sleep(10);
		// Select PARAM_RD register
		// Read 1 byte of data
		status = (byte)device.read(0x2E);
		
		// Select PARAM_WR register
		// High Signal Range
		device.write(0x17, (byte)0x20);
		// Select COMMAND register
		// Select ALS_IR_ADC_MISC register in RAM
		device.write(0x18, (byte)(0x1F | 0xA0));
		Thread.sleep(10);
		// Select PARAM_RD register
		// Read 1 byte of data
		status = (byte)device.read(0x2E);
		
		// Select PARAM_WR register
		// Set 511 ADC Clock
		device.write(0x17, (byte)0x70);
		// Select COMMAND register
		// Select ALS_VIS_ADC_COUNTER register in RAM
		device.write(0x18, (byte)(0x10 | 0xA0));
		Thread.sleep(10);
		// Select PARAM_RD register
		// Read 1 byte of data
		status = (byte)device.read(0x2E);
		
		// Select PARAM_WR register
		// High Signal Range
		device.write(0x17, (byte)0x20);
		// Select COMMAND register
		// Select ALS_VIS_ADC_MISC register in RAM
		device.write(0x18, (byte)(0x12 | 0xA0));
		Thread.sleep(10);
		// Select PARAM_RD register
		// Read 1 byte of data
		status = (byte)device.read(0x2E);
		
		// Select COMMAND register
		// Start ALS conversion
		device.write(0x18, (byte)0x0E);
		Thread.sleep(500);
		
		// Read 4 bytes from address 0x22(34)
		// visible lsb, visible msb, ir lsb, ir msb
		byte[] data = new byte[4];
		device.read(0x22, data, 0, 4);
		
		// Convert the data
		int visible = ((data[1] & 0xFF) * 256 + (data[0] & 0xFF));
		int ir = ((data[3] & 0xFF) * 256 + (data[2] & 0xFF));
		
		// Read 2 bytes from address 0x2C(44)
		// uv lsb, uv msb
		device.read(0x2C, data, 0, 2);
		
		// Convert the data
		int uv = ((data[1] & 0xFF) * 256 + (data[0] & 0xFF));
		
		// Output data to screen
		System.out.printf("Ambient Light Of Source : %d lux %n", visible);
		System.out.printf("IR Of Source : %d lux %n", ir);
		System.out.printf("UV Of the Source : %d lux %n", uv);
	}
}	

File: Python/si1132.py
# Distributed with a free-will license.
# Use it any way you want, profit or free, provided it fits in the licenses of its associated works.
# SI1132
# This code is designed to work with the SI1132_I2CS I2C Mini Module available from ControlEverything.com.
# https://www.controleverything.com/content/Light?sku=SI1132_I2CS#tabs-0-product_tabset-2

#import usmbus as smbus
#import smbus
import time
from machine import Pin

class SI1132:
    def __init__(
                 self,
                 smbus=None):
        self.bus = smbus
    
    def read(self):
        # SI1132 address, 0x60(96)
        # Enable UVindex measurement coefficients
        self.bus.write_byte_data(0x60, 0x13, 0x29)
        self.bus.write_byte_data(0x60, 0x14, 0x89)
        self.bus.write_byte_data(0x60, 0x15, 0x02)
        self.bus.write_byte_data(0x60, 0x16, 0x00)

        # SI1132 address, 0x60(96)
        # Select PARAM_WR register, 0x17(23)
        # 		0xF0(15)	Enable UV, Visible, IR
        self.bus.write_byte_data(0x60, 0x17, 0xF0)
        # SI1132 address, 0x60(96)
        # Select COMMAND register, 0x18(24)
        #		0x01 | 0xA0(161)   Select CHLIST register in RAM
        self.bus.write_byte_data(0x60, 0x18, (0x01 | 0xA0))
        time.sleep(0.01)
        # SI1132 address, 0x60(96)
        # Read data back from 0x2E(46), 1 byte
        response = self.bus.read_byte_data(0x60, 0x2E)

        # SI1132 address, 0x60(96)
        # Select INT Output Enable register, 0x03(03)
        #		0x01(01)	INT pin driven low
        self.bus.write_byte_data(0x60, 0x03, 0x01)
        # SI1132 address, 0x60(96)
        # Select ALS Interrupt Enable register, 0x04(04)
        #		0x01(01)	Assert INT pin whenever VIS or UV measurements are ready
        self.bus.write_byte_data(0x60, 0x04, 0x01)
        # SI1132 address, 0x60(96)
        # Select HW_KEY register, 0x07(07)
        #		0x17(23)	Default value
        self.bus.write_byte_data(0x60, 0x07, 0x17)

        # SI1132 address, 0x60(96)
        # Select PARAM_WR register, 0x17(23)
        #		0x00(0)		Small IR photodiode
        self.bus.write_byte_data(0x60, 0x17, 0x00)
        # SI1132 address, 0x60(96)
        # Select COMMAND register, 0x18(24)
        #		0x0E | 0xA0(174)    Select ALS_IR_ADCMUX register in RAM
        self.bus.write_byte_data(0x60, 0x18, (0x0E | 0xA0))
        time.sleep(0.01)
        # SI1132 address, 0x60(96)
        # Read data back from 0x2E(46), 1 byte
        response = self.bus.read_byte_data(0x60, 0x2E)

        # SI1132 address, 0x60(96)
        # Select PARAM_WR register, 0x17(23)
        #		0x00(0)		Set ADC Clock divided / 1
        self.bus.write_byte_data(0x60, 0x17, 0x00)
        # SI1132 address, 0x60(96)
        # Select COMMAND register, 0x18(24)
        #		0x1E | 0xA0(190)    Select ALS_IR_ADC_GAIN register in RAM
        self.bus.write_byte_data(0x60, 0x18, (0x1E | 0xA0))
        time.sleep(0.01)
        # SI1132 address, 0x60(96)
        # Read data back from 0x2E(46), 1 byte
        response = self.bus.read_byte_data(0x60, 0x2E)

        # SI1132 address, 0x60(96)
        # Select PARAM_WR register, 0x17(23)
        # 		0x70(112)	Set 511 ADC Clock
        self.bus.write_byte_data(0x60, 0x17, 0x70)
        # SI1132 address, 0x60(96)
        # Select COMMAND register, 0x18(24)
        #		0x1D | 0xA0(189)    Select ALS_IR_ADC_COUNTER register in RAM
        self.bus.write_byte_data(0x60, 0x18, (0x1D | 0xA0))
        time.sleep(0.01)
        # SI1132 address, 0x60(96)
        # Read data back from 0x2E(46), 1 byte
        response = self.bus.read_byte_data(0x60, 0x2E)

        # SI1132 address, 0x60(96)
        # Select PARAM_WR register, 0x17(23)
        # 		0x00(0)		Set ADC Clock divided / 1
        self.bus.write_byte_data(0x60, 0x17, 0x00)
        # SI1132 address, 0x60(96)
        # Select COMMAND register, 0x18(24)
        #		0x11 | 0xA0(177)    Select ALS_VIS_ADC_GAIN register in RAM
        self.bus.write_byte_data(0x60, 0x18, (0x11 | 0xA0))
        time.sleep(0.01)
        # SI1132 address, 0x60(96)
        # Read data back from 0x2E(46), 1 byte
        response = self.bus.read_byte_data(0x60, 0x2E)

        # SI1132 address, 0x60(96)
        # Select PARAM_WR register, 0x17(23)
        #		0x20(32)	High Signal Range
        self.bus.write_byte_data(0x60, 0x17, 0x20)
        # SI1132 address, 0x60(96)
        # Select COMMAND register, 0x18(24)
        #		0x1F | 0xA0(191)    Select ALS_IR_ADC_MISC register in RAM
        self.bus.write_byte_data(0x60, 0x18, (0x1F | 0xA0))
        time.sleep(0.01)
        # SI1132 address, 0x60(96)
        # Read data back from 0x2E(46), 1 byte
        response = self.bus.read_byte_data(0x60, 0x2E)

        # SI1132 address, 0x60(96)
        # Select PARAM_WR register, 0x17(23)
        # 		0x70(112)	Set 511 ADC Clock
        self.bus.write_byte_data(0x60, 0x17, 0x70)
        # SI1132 address, 0x60(96)
        # Select COMMAND register, 0x18(24)
        #		0x10 | 0xA0(176)    Select ALS_VIS_ADC_COUNTER register in RAM
        self.bus.write_byte_data(0x60, 0x18, (0x10 | 0xA0))
        time.sleep(0.01)
        # SI1132 address, 0x60(96)
        # Read data back from 0x2E(46), 1 byte
        response = self.bus.read_byte_data(0x60, 0x2E)

        # SI1132 address, 0x60(96)
        # Select PARAM_WR register, 0x17(23)
        #		0x20(32)	High Signal Range
        self.bus.write_byte_data(0x60, 0x17, 0x20)
        # SI1132 address, 0x60(96)
        # Select COMMAND register, 0x18(24)
        #		0x12 | 0xA0(178)    Select ALS_VIS_ADC_MISC register in RAM
        self.bus.write_byte_data(0x60, 0x18, (0x12 | 0xA0))
        time.sleep(0.01)
        # SI1132 address, 0x60(96)
        # Read data back from 0x2E(46), 1 byte
        response = self.bus.read_byte_data(0x60, 0x2E)

        # SI1132 address, 0x60(96)
        # Select COMMAND register, 0x18(24)
        #		0x0E(14)	Start ALS conversion
        self.bus.write_byte_data(0x60, 0x18, 0x0E)
        time.sleep(0.5)

        # SI1132 address, 0x60(96)
        # Read data back from 0x22(34), 4 bytes
        # visible lsb, visible msb, ir lsb, ir msb
        data = self.bus.read_i2c_block_data(0x60, 0x22, 4)

        # Convert the data
        visible = data[1] * 256 + data[0]
        ir = data[3] * 256 + data[2]

        # SI1132 address, 0x60(96)
        # Read data back from 0x2C(44), 2 bytes
        # uv lsb, uv msb
        data = self.bus.read_i2c_block_data(0x60, 0x2C, 2)

        # Convert the data
        uv = data[1] * 256 + data[0]

        # Output data to screen
        #print("Visible Light of Source : %d lux" %visible)
        #print("IR Of Source : %d lux" %ir)
        #print("UV Of the Source : %d lux" %uv)
        print('si1132 uv, ir, visible = %d, %d, %d lux' % (uv, ir, visible))
        return uv, ir, visible
